tests/mod.nu

```diff
@@ -147,3 +147,28 @@ export def env-vars-are-set [] {
     assert equal $env.NUPM_CACHE $dirs.DEFAULT_NUPM_CACHE
     assert equal $env.NUPM_REGISTRIES $dirs.DEFAULT_NUPM_REGISTRIES
 }
+
+export def generate-local-registry [] {
+    with-test-env {
+        # cp -r tests/packages $env.NUPM_TEMP
+        mkdir ($env.NUPM_TEMP | path join packages registry)
+
+        let reg_file = [tests packages registry registry.nuon] | path join
+        let tmp_reg_file = [
+            $env.NUPM_TEMP packages registry test_registry.nuon
+        ]
+        | path join
+
+        touch $tmp_reg_file
+
+        [spam_script spam_script_old spam_custom spam_module] | each {|pkg|
+            cd ([tests packages $pkg] | path join)
+            nupm publish $tmp_reg_file --local --save --path $'../($pkg)'
```

Comment:
would the following be more robust on Windows? (yeah i'm always worried about Windows lol)
```suggestion
            nupm publish $tmp_reg_file --local --save --path (".." | path join $pkg)
```